### PR TITLE
Point to right directory where Hycom data is stored

### PR DESCRIPTION
**Pull request by Mikejmnez**
## Do not merge yet. 
There are a couple of things that need fixing:

1) **Data for testing** I can provide with a test file with HyCOM data output, and a test for opening / creating the ospy object with such data.

2) **Decode variables**. This is an issue that I have been having with this dataset when trying to make simple plots. Right now, the dataset is created with the xarray argument `decode_cf=False` (see intake catalog). As a result, both the time and variables do not have the proper format, e.g. salinity and temperature are not corrected for `scale_factor`, `offset_value` and `missing_value` . The problem with `decode_cf=True` when creating the dataset (it usually corrects this scale and offset vals) is that it gives an error because the `time` has 'encoding' that xarray doesn't like...

setting `decode_cf=True` when creating the dataset yields 

```python
ValueError: Unable to parse date string 'analysis'
During handling of the above exception, another exception occurred:

ValueError: unable to decode time units 'hours since analysis' with 'the default calendar'. 
Try opening your dataset with decode_times=False or installing cftime if it is not installed.
```

if I manually create the dataset with `decode_cf=False` and then do

```python
ds = xr.decode_cf(ds, use_cftime=True)
```

I get the same error

```python
ValueError: unable to decode time units 'hours since analysis' with 'the default calendar'. 
Try opening your dataset with decode_times=False or installing cftime if it is not installed.
```

If we leave `decode_cf=False`, the time variable is an array of floats64

```python
time 1.731e+05 1.731e+05 ... 1.741e+05
```

I think this error is related to that in https://github.com/pydata/xarray/issues/521

do you have a suggestion for decoding the dataset correctly when reading from intake catalog?

Reading the dataset with `decode_cf=False` and doing ```ds.time``` yields

```python
long_name :    Valid Time
units :    hours since 2000-01-01 00:00:00
time_origin :    2000-01-01 00:00:00
calendar :    gregorian
axis :    T
```

### CODE REVIEW

**Comment by ThomasHaine** [comment]
Thanks Miguel. @malmans2 can you comment on the data formatting issue?

**Comment by malmans2** [comment]
I don't think the `time` coordinate is the issue. Maybe there's another variable with attributes that are not CF compliant? The attributes of `time` look in good shape, so there should be another variable with attribute `hours since analysis`.

If that's the case, you have a few options:
1. If the variables creating issues are not useful, ignore them using the `drop_variables` argument of `open_dataset`
2. Edit the file attributes and make them CF-compliant (i.e., replace hours since analysis with hours since YYYY-MM-DD hh:mm:ss)
3. If you don't want to edit the raw files, use the `preprocess` argument of `open_dataset` and update the attributes while reading
4. If there's no way to make the attributes CF-compliant, use `preprocess` to decide which variable decode (i.e., apply `decode_cf` only to a subset of variables

**Comment by malmans2** [comment]
BTW, I don't think it would work in your case, but I often use the argument `decode_timedelta=False` to avoid problems with variables such as ice age. It's easy to test though!

**Comment by Mikejmnez** [comment]
All good suggestions. I'll work on these and report back. Thanks @malmans2 !

**Comment by Mikejmnez** [comment]
Ok, figured out what variable was causing trouble (`tau`, which only has dimensions of time). Dropped it and now I can load the dataset while setting `Decode_cf=True`.

@malmans2 Now that we can proceed with this pull request, I need to send you a test dataset (nc) along with its intake file, right? This in order to add a test that it can open and create the ospy object.

The location and format (e.g. single zarr file) will likely change, but this should be merged in the next couple of days after the changes are done...

**Comment by malmans2** [comment]
> @malmans2 Now that we can proceed with this pull request, I need to send you a test dataset (nc) along with its intake file, right? This in order to add a test that it can open and create the ospy object.

Yes, but maybe it's better if we move the data in yours (or @asiddi24 @ThomasHaine @renskegelderloos ) onedrive? So basically add your data, move to another onedrive, and change the url here: https://github.com/hainegroup/oceanspy/blob/master/oceanspy/tests/conftest.py

**Comment by Mikejmnez** [comment]
> Yes, but maybe it's better if we move the data in yours (or @asiddi24 @ThomasHaine @renskegelderloos ) onedrive? So basically add your data, move to another onedrive, and change the url here: https://github.com/hainegroup/oceanspy/blob/master/oceanspy/tests/conftest.py

I agree. For the time being we can move it to my onedrive and figure out later a more stable long-term location.

**Comment by Mikejmnez** [comment]
@malmans2 I am getting this error where during testing `oceanspy` can't find the `hycom_test.nc` file. The error is in the `.yaml` file I sent you. I think the correct location should be

```python
urlpath: ./oceanspy/tests/Data/hycom_test.nc
```
could you fixed it on the `hycom_test.yaml` file?

File: oceanspy/tests/conftest.py [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-        "EVtxjAQL13tCt7dFHNrzsrwBuqdjDe3zrvrJ625YrjrF0g"
-        "?download=1",
-        "tar xvzf Data.tar.gz",
-        "rm -f Data.tar.gz",
-    ]
-    subprocess.call("&&".join(commands), shell=True)
+    fnames = pooch.retrieve(
+        url="https://zenodo.org/record/5825166/files/Data.tar.gz?download=1",
+        processor=Untar(),
+        known_hash="165bb4c0459a3b776efe9f94e8f7843140bc7e90b9d686af574cdb8c11006ba2",
```
Comment (malmans2) [inline oceanspy/tests/conftest.py]:
```suggestion
        url="https://zenodo.org/record/5832607/files/Data.tar.gz?download=1",
        processor=Untar(),
        known_hash="98b2bfadefa62dd223224c797354f9266b54143c2af3c4b6fe676d8547e7d5ee",
```

**Comment by malmans2** [comment]
@Mikejmnez Done!

**Comment by Mikejmnez** [comment]
> @Mikejmnez Done!

Awesome! If everything looks good, you can merge now. I figured what was causing trouble with ```decode_cf=True```, the variable `tau` which is only time dependent and now important  for now so it has been removed.

**Comment by malmans2** [comment]
Great! Merging...

**Review by malmans2 — APPROVED** [review]

**Pull request merged**